File: src/java/magma/app/Primitive.java
package magma.app;

import magma.api.Type;

public enum Primitive implements Type {
    String("string"),
    Number("number"),
    Var("var"),
    Void("void"),
    Unknown("unknown"),
    I8("I8"),
    I32("I32");

    private final String value;

    Primitive(final String value) {
        this.value = value;
    }

    @Override
    public String generate() {
        return this.value;
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return Primitive.Var == this;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }
}

File: src/node/magma/app/Primitive.ts
import { Type } from "../../magma/api/Type";
export class Primitive implements Type {
	static String: Primitive = new Primitive("string");
	static Number: Primitive = new Primitive("number");
	static Var: Primitive = new Primitive("var");
	static Void: Primitive = new Primitive("void");
	static Unknown: Primitive = new Primitive("unknown");
	static I8: Primitive = new Primitive("I8");
	static I32: Primitive = new Primitive("I32");
	value: string;
	constructor (value: string) {
		this.value = value;
	}
	generate(): string {
		return this.value;
	}
	isFunctional(): boolean {
		return false;
	}
	isVar(): boolean {
		return Primitive.Var === this;
	}
	generateBeforeName(): string {
		return "";
	}
}
